### PR TITLE
Review default

### LINKED ISSUE #5664: Localization - Do not update translations until default is reviewed
Following today's large merges, some changes need to be made to `rtdata/languages/default` and possibly relevant keys in source files. Please do not make translations until this issue is closed.

### PR DESCRIPTION
**Pull request by Beep6581**
I'd like to merge ASAP to allow branching off for 5.9.

### CODE REVIEW

File: rtdata/languages/default [hunk 1]
```diff
@@ -137,7 +138,6 @@ FILEBROWSER_PARTIALPASTEPROFILE;Paste - partial
 FILEBROWSER_PASTEPROFILE;Paste
 FILEBROWSER_POPUPCANCELJOB;Cancel job
 FILEBROWSER_POPUPCOLORLABEL;Color label
-FILEBROWSER_POPUPCOLORLABEL0;Label: None
```
Comment (Thanatomanic) [inline rtdata/languages/default]: I don't think this needs to be deleted. See https://github.com/Beep6581/RawTherapee/blob/2924128311d7fa5165a477b83a84c57870c3799b/rtgui/filebrowser.cc#L220-L222

File: rtdata/languages/default [hunk 2]
```diff
@@ -155,7 +155,6 @@ FILEBROWSER_POPUPPROCESS;Put to queue
 FILEBROWSER_POPUPPROCESSFAST;Put to queue (Fast export)
 FILEBROWSER_POPUPPROFILEOPERATIONS;Processing profile operations
 FILEBROWSER_POPUPRANK;Rank
-FILEBROWSER_POPUPRANK0;Unrank
```
Comment (Thanatomanic) [inline rtdata/languages/default]: Same, should not be deleted afaik

File: rtdata/languages/default [hunk 3]
```diff
@@ -1286,9 +1285,6 @@ HISTORY_MSG_1061;Local - CIECAM Source absolute
 HISTORY_MSG_1062;Local - CIECAM Surround Source
 HISTORY_MSG_1063;Local - CIECAM Saturation
 HISTORY_MSG_1064;Local - CIECAM Chroma
-HISTORY_MSG_1062;Local - CIECAM Lightness
```
Comment (Thanatomanic) [inline rtdata/languages/default:1289]: Why do these get deleted? This exists https://github.com/Beep6581/RawTherapee/blob/0a554282054c061fb06422ebae1f218fec8003e2/rtengine/procevents.h#L1092 and is implemented here https://github.com/Beep6581/RawTherapee/blob/2924128311d7fa5165a477b83a84c57870c3799b/rtgui/locallabtools2.cc#L10001

File: rtdata/languages/default [hunk 4]
```diff
@@ -3125,7 +3127,6 @@ TP_LOCALLAB_MERNIN;Screen
 TP_LOCALLAB_MERONE;Normal
 TP_LOCALLAB_MERSAT;Saturation
 TP_LOCALLAB_MERSEV;Soft Light (legacy)
-TP_LOCALLAB_MERSEV0;Soft Light Illusion
```
Comment (Thanatomanic) [inline rtdata/languages/default]: Probably should not be deleted

File: rtdata/languages/default (hunk 1 above)
Comment (Lawrence37) [reply rtdata/languages/default]: It's probably due to this bug: https://github.com/Beep6581/RawTherapee/pull/6503#issuecomment-1172974778
If there are two identical keys except one has an added `0` at the end, the one with the zero gets removed.

File: rtdata/languages/default (hunk 3 above)
Comment (Lawrence37) [reply rtdata/languages/default:1289]: It's a duplicate key. We need to check which string is correct.

File: rtdata/languages/default (hunk 1 above)
Comment (Beep6581) [reply rtdata/languages/default]: I did not delete that :confused:

**Comment by Beep6581** [comment]
@Thanatomanic @Lawrence37 thank you, good catches! Looks like `generateTranslationDiffs` is deleting keys it should not be deleting cccb9f543c06546847e613a757463e9aeccba912. I confirm the bug with `sort` described in #6503. It could also be that the script is written for Bash, while I'm now using ZSH in Manjaro, where parameter expansion is not 100% compatible with Bash's implementation.

I was recently thinking of rewriting those scripts in Python, for the fun of it... now there's a good reason to do that.

Anyhow I'll go through the changes and restore the missing keys tomorrow.

**Comment by Thanatomanic** [comment]
> I was recently thinking of rewriting those scripts in Python, for the fun of it... now there's a good reason to do that.

@Entropy512 had been doing some Python coding to find unused keys (https://github.com/Beep6581/RawTherapee/issues/6457) and finding orphaned keys (https://github.com/Beep6581/RawTherapee/pull/6458 - merged). I am pretty sure he would be willing to cook up a replacement for `generateTranslationDiffs` in Python.

**Comment by Entropy512** [comment]
Yup, although as I mentioned in the PMs over on pixls - do we even want to go that route longterm, vs take on moving to something like gettext as a post-5.9 task?

What's the history on the homegrown engine here - was it just that gettext was not around back in the day, and/or an artifact of the days before RT was open sourced and perhaps there would have been a licensing conflict that no longer exists?

A lot of tooling exists around gettext vs. managing our own - for example pixls has an instance of weblate that darktable uses for translations

File: rtdata/languages/default (hunk 3 above)
Comment (Lawrence37) [reply rtdata/languages/default:1289]: The events match the strings that are still here so I think this is correct.
https://github.com/Beep6581/RawTherapee/blob/0a554282054c061fb06422ebae1f218fec8003e2/rtengine/procevents.h#L1091-L1093

File: rtdata/languages/default [hunk 5]
```diff
@@ -4088,3 +4082,5 @@ ZOOMPANEL_ZOOMFITCROPSCREEN;Fit crop to screen\nShortcut: <b>f</b>
 ZOOMPANEL_ZOOMFITSCREEN;Fit whole image to screen\nShortcut: <b>Alt</b>-<b>f</b>
 ZOOMPANEL_ZOOMIN;Zoom In\nShortcut: <b>+</b>
 ZOOMPANEL_ZOOMOUT;Zoom Out\nShortcut: <b>-</b>
+//TP_LOCALLAB_SHOWMASKTYP_TOOLTIP;Mask and modifications can be chosen.\nBlur and noise : in this case it is not used for 'denoise'.\nDenoise : in this case it is not used for 'blur and noise'.\n\nBlur and noise + denoise : mask is shared, be careful to 'show modifications' and 'scope'.
```
Comment (Lawrence37) [inline rtdata/languages/default]: This can be removed. There is another `TP_LOCALLAB_SHOWMASKTYP_TOOLTIP` with a better message.

**Comment by Beep6581** [comment]
I had to do some unexpected work this weekend and haven't had time to address this, I'll do that tomorrow.

**Comment by Beep6581** [comment]
@Entropy512 

> Yup, although as I mentioned in the PMs over on pixls - do we even want to go that route longterm, vs take on moving to something like gettext as a post-5.9 task?
> 
> What's the history on the homegrown engine here - was it just that gettext was not around back in the day, and/or an artifact of the days before RT was open sourced and perhaps there would have been a licensing conflict that no longer exists?
> 
> A lot of tooling exists around gettext vs. managing our own - for example pixls has an instance of weblate that darktable uses for translations

gettext tools were garbage back in the day, a nightmare to use - QtLinguist was way better and there was hope of porting RT to Qt. But hope for the Qt port fuzzed out and the gettext tools improved several years ago already, so RT would benefit from using gettext.

File: rtdata/languages/default (hunk 3 above)
Comment (Desmis) [reply rtdata/languages/default:1289]: I looked at both, the code in "locallabtools2.cc", "procevents.h", "refreshmap.cc" and what appears in "history". There is duplication in "HISTORY_MSG"

The good code is
HISTORY_MSG_1061;Local - CIECAM Source absolute
HISTORY_MSG_1062;Local - CIECAM Surround Source
HISTORY_MSG_1063;Local - CIECAM Saturation
HISTORY_MSG_1064;Local - CIECAM Chroma
!HISTORY_MSG_1062;Local - CIECAM Lightness
!HISTORY_MSG_1063;Local - CIECAM Brightness
HISTORY_MSG_1064;Local - CIECAM threshold
HISTORY_MSG_1065;Local - CIECAM lightness J
HISTORY_MSG_1066;Local - CIECAM brightness
HISTORY_MSG_1067;Local - CIECAM Contrast J

So you have to delete the 2 lines with a !
!HISTORY_MSG_1062;Local - CIECAM Lightness
!HISTORY_MSG_1063;Local - CIECAM Brightness

jacques

File: rtdata/languages/default (hunk 4 above)
Comment (Desmis) [reply rtdata/languages/default]: I confirm 
TP_LOCALLAB_MERSEV0;Soft Light Illusion
Must not be deleted

Jacques

Comment (Beep6581) [reply rtdata/languages/default]: Restored.

File: rtdata/languages/default (hunk 1 above)
Comment (Beep6581) [reply rtdata/languages/default]: Restored.

File: rtdata/languages/default (hunk 2 above)
Comment (Beep6581) [reply rtdata/languages/default]: Restored.

File: rtdata/languages/default (hunk 5 above)
Comment (Beep6581) [reply rtdata/languages/default]: Removed.

File: rtdata/languages/default (hunk 3 above)
Comment (Beep6581) [reply rtdata/languages/default:1289]: Duplicates were already deleted, nothing to do.

**Comment by Beep6581** [comment]
Ready for merge. I will merge tonight if no objections.

**Pull request merged**